spec/logger.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import * as logger from '../src/logger';
+
+const SUPPORTS_STRUCTURED_LOGS =
+  parseInt(process.versions?.node?.split('.')?.[0] || '8', 10) >= 10;
```

Comment:
This test is essentially relying on the runtime that the test is running in. This is a bit odd to be since the test behaves differently depending on which runtime it's running in. It would be better to stub `process.versions.node` so you can write separate tests for Node versions <10 and >=10

Reply:
There's no easy way to stub the behavior since it happens before anything is actually defined or exported (I tried). This was what I landed on and I think to some extent it makes sense -- it properly tests behaviors in each runtime and our CI tests do run the Node 8 tests.

Reply:
Bummer. Ok this works then.